osu.Desktop/Updater/SimpleUpdateManager.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.IO.Network;
+using osu.Framework.Platform;
+using osu.Game;
+using osu.Game.Graphics;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+
+namespace osu.Desktop.Updater
+{
+    /// <summary>
+    /// An update manager that shows notifications if a newer release is detected.
+    /// Installation is left up to the user.
+    /// </summary>
+    internal class SimpleUpdateManager : CompositeDrawable
+    {
+        private NotificationOverlay notificationOverlay;
+        private string version;
+        private GameHost host;
+
+        [BackgroundDependencyLoader]
+        private void load(NotificationOverlay notification, OsuGameBase game, GameHost host)
+        {
+            notificationOverlay = notification;
+
+            this.host = host;
+            version = game.Version;
+
+            if (game.IsDeployedBuild)
+                Schedule(() => Task.Run(() => checkForUpdateAsync()));
```

Comment:
Is there a reason for this schedule? (`notificationOverlay.Post()` is internally scheduled, which is thread-safe) Won't this essentially block the update thread?

Reply:
I copied this from the squirrel updater. It's simply there to delay checking until after the first frame of the game runs, to delay it later into execution.

What do you mean by blocking the update thread? It's also run as a task inside, so this shouldn't happen.

Reply:
Ah right, okay.